test/IISIntegration.FunctionalTests/Utilities/IISApplication.cs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNetCore.Server.IntegrationTesting.Common;
+using Microsoft.Extensions.Logging;
+using Microsoft.Web.Administration;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting
+{
+    internal class IISApplication
+    {
+        private readonly ServerManager _serverManager = new ServerManager();
+        private readonly DeploymentParameters _deploymentParameters;
+        private readonly ILogger _logger;
+        private string _originalAspNetCoreDllLocation = "";
+
+        public IISApplication(DeploymentParameters deploymentParameters, ILogger logger)
+        {
+            _deploymentParameters = deploymentParameters;
+            _logger = logger;
+            WebSiteName = CreateTestSiteName();
+            AppPoolName = $"{WebSiteName}Pool";
+        }
+
+        public string WebSiteName { get; }
+
+        public string AppPoolName { get; }
+
+        public void StartIIS(Uri uri, string contentRoot)
+        {
+            using (_logger.BeginScope("StartIIS"))
+            {
+                var port = uri.Port;
+                if (port == 0)
+                {
+                    port = (uri.Scheme == "https") ? TestPortHelper.GetNextSSLPort() : TestPortHelper.GetNextPort();
+                }
+
+                ConfigureAppPool();
+
+                ConfigureSite(contentRoot, port);
+
+                ConfigureAppHostConfig(contentRoot);
+
+                _serverManager.CommitChanges();
+            }
+        }
+
+        private void ConfigureAppPool()
+        {
+            var pool = _serverManager.ApplicationPools.Add(AppPoolName);
+            pool.StartMode = StartMode.AlwaysRunning;
+            pool.ProcessModel.IdentityType = ProcessModelIdentityType.LocalSystem;
+            pool.ProcessModel.LoadUserProfile = true;
+        }
```

Comment:
[Docs](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/?view=aspnetcore-2.1&tabs=aspnetcore2x#create-the-iis-site) recommend setting the CLR version to "No Managed Code".  I think we should do the same, by setting `pool.ManagedRuntimeVersion = string.Empty;`